src/app/shared/components/add-record/add-record.component.ts

```diff
@@ -114,12 +114,22 @@ export class AddRecordComponent implements OnInit, OnDestroy {
   }
 
   private createProof() {
-    return this.proofService.createProof()
+    console.log('enableGeolocation',this.proofService.enableGeolocation)
+    if ( this.proofService.enableGeolocation == 'enable') {
+      return this.proofService.createProof()
       .pipe(
         tap(proof => this.updateRecordProof(proof)),
         tap(() => this.proofStatus = ProofStatus.COMPLETE),
         takeUntil(this.destroy$),
       );
+    } else {
+      return this.proofService.createDisableLocationProof()
+      .pipe(
+        tap(proof => this.updateRecordProof(proof)),
+        tap(() => this.proofStatus = ProofStatus.COMPLETE),
+        takeUntil(this.destroy$),
+      );
+    }
```

Comment:
```suggestion
    } else {
      const proof = this.proofService.createProofWithoutLocation();
      this.updateRecordProof(proof);
      this.proofStatus = ProofStatus.COMPLETE);
    }
```

No need to use observable here